modules/kit-elements-defaults/controller.php

```diff
@@ -0,0 +1,242 @@
+		);
+
+		return new \WP_REST_Response( [], 201 );
+	}
+
+
+	/**
+	 * DELETE '/kit-elements-defaults/{type}'
+	 */
+	private function register_destroy() {
+		register_rest_route( static::NAMESPACE, static::BASE_URL . '/(?P<type>[\w\-_]+)', [
+			[
+				'methods' => 'DELETE',
+				'args' => [
+					'type' => [
+						'type' => 'string',
+						'description' => 'The type of the element.',
+						'required' => true,
+						'validate_callback' => function( $param ) {
+							return $this->validate_type( $param );
+						},
+					],
+				],
+				'callback' => $this->wrap_callback( function ( \WP_REST_Request $request ) {
+					return $this->destroy( $request );
+				} ),
+				'permission_callback' => function () {
+					return current_user_can( 'manage_options' );
+				},
+			],
+		] );
+	}
+
+	private function destroy( \WP_REST_Request $request ) {
+		$kit = Plugin::$instance->kits_manager->get_active_kit();
+
+		$data = $kit->get_json_meta( static::META_KEY );
+
+		unset( $data[ $request->get_param( 'type' ) ] );
+
+		$kit->update_meta(
+			static::META_KEY,
+			wp_json_encode( $data )
+		);
+
+		return new \WP_REST_Response( [], 204 );
+	}
+
+	/**
+	 * Wrapper to validate that the kit is valid.
+	 * Essentially behaves as a middleware.
+	 *
+	 * @param $callback
+	 *
+	 * @return \Closure
+	 */
+	private function wrap_callback( $callback ) {
+		return function ( \WP_REST_Request $request ) use ( $callback ) {
+			$kit = Plugin::$instance->kits_manager->get_active_kit();
+
+			// TODO: Is this right?
+			$is_valid_kit = $kit && $kit->get_main_id();
+
+			if ( ! $is_valid_kit ) {
+				return new \WP_Error( 'invalid_kit', 'Kit doesn\'t exist.', [
+					'status' => 404,
+				] );
+			}
+
+			return $callback( $request );
+		};
+	}
+
+	private function validate_type( $param ) {
+		$element_types = array_keys( Plugin::$instance->elements_manager->get_element_types() );
+		$widget_types = array_keys( Plugin::$instance->widgets_manager->get_widget_types() );
+
+		return in_array(
+			$param,
+			array_merge( $element_types, $widget_types ),
+			true
+		);
+	}
+
+	private function sanitize_settings( $param, \WP_REST_Request $request ) {
+		$element = $this->get_element_instance( $request->get_param( 'type' ) );
+
+		if ( ! $element ) {
+			return [];
+		}
+
+		$sanitized_settings = $this->remove_invalid_settings( $element, $param );
+		$sanitized_settings = $this->remove_secret_settings( $element, $sanitized_settings );
+
+		return $sanitized_settings;
+	}
+
+	private function remove_invalid_settings( Element_Base $element, $settings ) {
+		$valid_element_controls = $element->get_controls();
+
+		// Remove the controls that don't exist in the element.
+		return array_intersect_key(
+			$settings,
+			$valid_element_controls
+		);
+	}
+
+	private function remove_secret_settings( Element_Base $element, $settings ) {
+		// We rely on the `on_export` method that should remove secrets from the settings.
+		if ( method_exists( $element, 'on_export' ) ) {
```

Comment:
Please check:
`\Elementor\Core\Base\Document::process_element_import_export`
Seems like it does more things then just call `on_export`, we should talk about it with @matipojo